tests/test.c

```diff
@@ -24,134 +24,39 @@
-        BUG();
-    }
-    else {
-        printk("String parameter parsing works!\n");
-    }
-
-    if (strcmp(opt_badstring, "tool")) {
-        printk("String parameter opt_badstring != tool: %s\n", opt_badstring);
-        BUG();
-    }
-    else {
-        printk("String parameter parsing works!\n");
-    }
-
-    if (opt_ulong != 42) {
-        printk("Integer parameter opt_ulong != 42: %d\n", opt_ulong);
-        BUG();
-    }
-    else {
-        printk("Integer parameter parsing works!\n");
-    }
-
-    if (!opt_bool || !opt_booltwo) {
-        printk("Boolean parameter opt_bool != true: %d\n", opt_bool);
-        printk("Boolean parameter opt_booltwo != true: %d\n", opt_booltwo);
-        BUG();
-    }
-    else {
-        printk("Boolean parameter parsing works!\n");
-    }
-
-    printk("\nMemmove testing:\n");
-    (void) memmove(memmove_string, opt_string, sizeof(opt_string));
-    if (!strcmp(memmove_string, opt_string)) {
-        printk("Moving around memory works!\n");
-    }
-    else {
-        printk("Memmove'ing did not work: %s (%p) != %s (%p)\n", memmove_string,
-               memmove_string, opt_string, opt_string);
-    }
-
-    src = (char *) range_string;
-    dst = (char *) range_string + 2;
-    (void) memmove(dst, src, 4);
-    if (!strcmp(range_string, "121234")) {
-        printk("Moving around memory with overlaping ranges works!\n");
-    }
-    else {
-        printk("Overlaping memmove'ing did not work: %s != %s\n", range_string, "121234");
-    }
-
-    cpu_freq_expect("Intel(R) Xeon(R) CPU E3-1270 V2 @ 3.50GHz", 3500000000);
-    cpu_freq_expect("Intel(R) Celeron(R) CPU J1900 @ 1.99GHz", 1990000000);
-    cpu_freq_expect("AMD G-T48L Processor", 0);
-    cpu_freq_expect("Intel(R) Atom(TM) CPU E3815 @ 1.46GHz", 1460000000);
-    cpu_freq_expect("Intel(R) Atom(TM) CPU E620 @ 600MHz", 600000000);
-    cpu_freq_expect("VIA C7 Processor 1000MHz", 1000000000);
-    cpu_freq_expect("Intel(R) Core(TM) i7 CPU 950 @ 3.07GHz", 3070000000);
-    cpu_freq_expect("AMD Ryzen Threadripper 1950X 16-Core Processor", 0);
-    cpu_freq_expect("Prototyp Amazing Foo One @ 1GHz", 1000000000);
-    cpu_freq_expect("Prototyp Amazing Foo Two @ 1.00GHz", 1000000000);
-
-    printk("Long mode to real mode transition:\n");
-    long_to_real();
+static const char opt_test_delims[] = ",";
+#include <cmdline.h>
 
-    return 0;
-}
-#else
-static int ktf_unit_tests(void) { return 0; }
-#endif
+static char opt_tests[128];
+string_cmd("tests", opt_tests);
 
 void test_main(void) {
-    printk("\nTest:\n");
-
-    ktf_unit_tests();
+    char *test_name, *opt;
+    int r;
+    unsigned n = 0;
+
+    printk("\nRunning tests\n");
+
+    opt = opt_tests;
+    while ((test_name = strtok(opt, opt_test_delims)) != NULL) {
+        test_fn *test = symbol_address(test_name);
+        if (!test)
+            panic("The test <%s> does not exist\n", test_name);
+        printk("Running: %s\n", test_name);
+        r = test(NULL);
```

Comment:
unrelated: next thing is to figure out to pass parameters to the tests, that would be super helpful!

Reply:
`param_$testname=...` maybe?